### ISSUE TITLE
v6.18.0+ crashes the Mac Catalyst apps when user interacts with Help menu

### ISSUE THREAD

**Issue by Czajnikowski** [bugsnag/bugsnag-cocoa#1484]
### Describe the bug
The app crashes when the user searches and hovers over available entries in the Help menu.

### Steps to reproduce
1. Open the Help menu
2. Write some search query
3. Hover the mouse over available entries

Check [the vid](
https://user-images.githubusercontent.com/973682/191205998-5eee3571-d093-4c01-8066-5a1350d04696.mov)

### Environment
* Bugsnag version: 6.18.0+
* CocoaPods version: We use SPM
* Carthage version (if any): We use SPM
* iOS/tvOS/macOS version(s): the app runs on Mac Catalyst, macOS 12.5, I'm pretty sure it's reproducible on other versions too
* Simulator or physical device: Physical device
* Xcode version: 14.0 (14A309), I'm pretty sure it's reproducible on other versions too
* Swift version (if applicable): I don't think it's applicable

### Example code snippet

I'm unsure if I can provide any code snippet, as the issue happens without any specific call to your SDK (likely some sizzling problem?). I am attaching a crash log for further investigation [Vectornator-2022-09-20-101415.txt](https://github.com/bugsnag/bugsnag-cocoa/files/9605478/Vectornator-2022-09-20-101415.txt).

**Comment by nickdowell** [comment]
Hi @Czajnikowski thanks for reporting, we're looking into this now 👍

<details><summary>Stacktrace:</summary>

```
0   CoreFoundation                      0x000000019f4d51a8 __exceptionPreprocess + 240
1   libobjc.A.dylib                     0x000000019f21fe04 objc_exception_throw + 60
2   CoreFoundation                      0x000000019f568f4c -[NSObject(NSObject) __retain_OA] + 0
3   CoreFoundation                      0x000000019f434554 ___forwarding___ + 1764
4   CoreFoundation                      0x000000019f433db0 _CF_forwarding_prep_0 + 96
5   Vectornator                         0x00000001048fc3e8 -[BSGNotificationBreadcrumbs addBreadcrumbForTableViewNotification:] + 96
6   CoreFoundation                      0x000000019f44951c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
7   CoreFoundation                      0x000000019f4edc18 ___CFXRegistrationPost_block_invoke + 52
8   CoreFoundation                      0x000000019f4edb84 _CFXRegistrationPost + 456
9   CoreFoundation                      0x000000019f4181c0 _CFXNotificationPost + 732
10  Foundation                          0x00000001a02ddc98 -[NSNotificationCenter postNotificationName:object:userInfo:] + 96
11  AppKit                              0x00000001a20e1f8c -[NSTableView _sendSelectionChangedNotificationForRows:columns:] + 244
12  AppKit                              0x00000001a20e143c -[NSTableView _doSelectIndexes:byExtendingSelection:indexType:funnelThroughSingleIndexVersion:] + 1788
13  AppKit                              0x00000001a20e0d1c -[NSTableView selectRowIndexes:byExtendingSelection:] + 136
14  Shortcut                            0x00000001d6ad7390 -[SCTMenuView mouseMoved:] + 388
15  AppKit                              0x00000001a21b11d4 forwardMethod + 200
16  AppKit                              0x00000001a21b11d4 forwardMethod + 200
17  AppKit                              0x00000001a21b11d4 forwardMethod + 200
18  AppKit                              0x00000001a21b11d4 forwardMethod + 200
19  AppKit                              0x00000001a21b0f18 -[NSTextView mouseMoved:] + 684
20  AppKit                              0x00000001a212be88 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 5908
21  AppKit                              0x00000001a212a50c -[NSWindow(NSEventRouting) sendEvent:] + 348
22  AppKit                              0x00000001a28d970c NSMenuWindowManagerMenuItemCarbonEventHandler + 4160
23  HIToolbox                           0x00000001a806a6c8 _ZL23DispatchEventToHandlersP14EventTargetRecP14OpaqueEventRefP14HandlerCallRec + 1084
24  HIToolbox                           0x00000001a8069b4c _ZL30SendEventToEventTargetInternalP14OpaqueEventRefP20OpaqueEventTargetRefP14HandlerCallRec + 356
25  HIToolbox                           0x00000001a807fe50 SendEventToEventTarget + 40
26  HIToolbox                           0x00000001a80aa638 _ZL29ToolboxEventDispatcherHandlerP25OpaqueEventHandlerCallRefP14OpaqueEventRefPv + 2520
27  HIToolbox                           0x00000001a806ab14 _ZL23DispatchEventToHandlersP14EventTargetRecP14OpaqueEventRefP14HandlerCallRec + 2184
28  HIToolbox                           0x00000001a8069b4c _ZL30SendEventToEventTargetInternalP14OpaqueEventRefP20OpaqueEventTargetRefP14HandlerCallRec + 356
29  HIToolbox                           0x00000001a807fe50 SendEventToEventTarget + 40
30  HIToolbox                           0x00000001a80fb3bc _ZL19IsUserStillTrackingP14MenuSelectDataPh + 3604
31  HIToolbox                           0x00000001a82392bc _ZL15TrackMenuCommonR14MenuSelectDataPhP13SelectionDataP10MenuResultS5_ + 1260
32  HIToolbox                           0x00000001a8106054 _ZL14MenuSelectCoreP8MenuData5PointdjPP13OpaqueMenuRefPt + 348
33  HIToolbox                           0x00000001a8105e48 _HandleMenuSelection2 + 416
34  AppKit                              0x00000001a21576f0 _NSHandleCarbonMenuEvent + 300
35  AppKit                              0x00000001a21574d0 _DPSEventHandledByCarbon + 68
36  AppKit                              0x00000001a1fb95b4 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 3280
37  AppKit                              0x00000001a1faafe0 -[NSApplication run] + 596
38  AppKit                              0x00000001a1f7c6fc NSApplicationMain + 1132
39  AppKit                              0x00000001a2252640 +[NSWindow _savedFrameFromString:] + 0
40  UIKitMacHelper                      0x00000001b4d173c0 UINSApplicationMain + 1276
41  UIKitCore                           0x00000001c74b4c2c UIApplicationMain + 164
42  Vectornator                         0x000000010484ee04 $s11Vectornator15ApplicationMainV4mainyyFZTf4d_n + 224
43  Vectornator                         0x00000001043502c0 main + 12
44  dyld                                0x0000000108c4908c start + 520
```

**Comment by nickdowell** [comment]
Fixed in [v6.23.1](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.23.1) 🎉

**Comment by Czajnikowski** [comment]
Works fine 👌 Thanks again @nickdowell 🙌
